src/ui/public/index_patterns/static_utils/index.js

```diff
@@ -0,0 +1,3 @@
+export function getFieldByName(fields, name) {
+  return fields.find(field => field.name === name);
```

Comment:
Feels a little over the top to abstract this out...

The name of this function could almost be automatically converted into its definition; that is to say I think it's almost never going to change and hides away how simple the definition actually is.